docs/search-views/semantic-search/concepts/distance-type.md

```diff
@@ -13,10 +13,104 @@ You have several distance type options. The different distance or similarity mea
 
 The L2 norm or Euclidean distance is probably the most commonly known metric. It represents the shortest straight line distance between two points in a space and is calculated as the square root of the sum of the squared differences between the corresponding elements of the two vectors.
 
+Euclidean distance incorporates both magnitude and direction of the vectors. It is sensitive to the scale and coordinates of the vectors. Euclidean distance is useful when the absolute distances between data points matter, rather than just orientation. For example, if searching for similar retail stores, Euclidean distance works well since proximity of stores geographically is important.
+
+A limitation of Euclidean distance is its sensitivity to scaling and noise. Vectors that are far apart in scale will have large distances even if oriented similarly. Outliers can also distort Euclidean distance.
+
+**Syntax:**
+
+```
+l2(vector1, vector2)
+```
+
+**Parameters:**
+
+| Key     | Type  | Description        |
+|---------|-------|-------------------------------|
+| vector1 | array | The first vector for comparison. |
+| vector2 | array | The second vector for comparison. |
+
+**Example:** Imagine a scenario where you're comparing two geographical points or coordinates. The L2 distance would give you the distance between them.
+
+*C8QL Sample - RETURN l2([10, 20, 30, 40], [5, 6, 7, 8]) -> 42.11*
+
 ## IP (Inner Product)
 
-The inner product (also known as the dot product) between two vectors is the sum of the products of their corresponding components. In the context of semantic search, inner product similarity is a measure of similarity that computes the dot product of the query vector and each document vector. High inner product values indicate high similarity. It's especially useful when dealing with vectors that represent things like user and item embeddings in recommendation systems.
+The inner product (also known as the dot product) between two vectors is the sum of the products of their corresponding components. In the context of semantic search, inner product similarity is a measure of similarity that computes the dot product of the query vector and each document vector. High inner product values indicate high similarity.
+
+Inner product incorporates both vector orientation through cosine similarity and vector magnitudes. Inner product is useful when scale matters, such as search based on weighted vectors. It works well for dense vectors. It's especially useful when dealing with vectors that represent things like user and item embeddings in recommendation systems.
+
+A limitation of inner product is sensitivity to noise and outliers, since large values are squared. Inner product also requires vectors of equal length.
```

Comment:
all the distance types need vectors of equal length; it is not unique to ip

Reply:
Is that a Macrometa limitation or a universal limitation? If it's just for Macrometa, we should mention it at the top. If it's a universal, then we may or may not mention it.

Reply:
its universal